### PR TITLE
Typo3 8 compatibility

### PR DESCRIPTION
**Pull request by anjeylink**
Support for TYPO3 8.4

### CODE REVIEW

File: Classes/Controller/EditorController.php [hunk 1]
```diff
@@ -54,6 +61,131 @@ class EditorController extends ActionController
...
+    {
+        /** @var BackendTemplateView $view */
+        parent::initializeView($view);
+
+        if ($this->view->getModuleTemplate() !== NULL) {
+            $pageRenderer = $this->view->getModuleTemplate()->getPageRenderer();
+            $pageRenderer->loadRequireJsModule('TYPO3/CMS/Themes/Colorpicker');
+            $pageRenderer->loadRequireJsModule('TYPO3/CMS/Themes/ThemesBackendModule');
+
+            $pageRenderer->addCssFile('../typo3conf/ext/themes/Resources/Public/Stylesheet/BackendModule.css');
```
Comment (MattiasNilsson) [inline Classes/Controller/EditorController.php]: Use a proper path resolver

File: Classes/Controller/EditorController.php [hunk 2]
```diff
@@ -54,6 +61,131 @@ class EditorController extends ActionController
...
+            $pageRenderer->addCssFile('../typo3conf/ext/themes/Resources/Public/Contrib/colorpicker/css/colorpicker.css');
```
Comment (MattiasNilsson) [inline Classes/Controller/EditorController.php]: Use a proper path resolver

File: Resources/Public/JavaScript/ThemesBackendModule.js [hunk 3]
```diff
@@ -55,11 +55,11 @@ define(['jquery'], function (jQuery) {
 		});
 	};
 
-	ThemesEditor.bindSaveIconEvent = function() {
+	/*ThemesEditor.bindSaveIconEvent = function() {
```
Comment (MattiasNilsson) [inline Resources/Public/JavaScript/ThemesBackendModule.js]: Remove commented code

File: Resources/Public/JavaScript/ThemesBackendModule.js [hunk 4]
```diff
@@ -21,7 +21,7 @@ define(['jquery'], function (jQuery) {
 		// toggle constant editor
 		ThemesEditor.bindEditToggleEvents();
 		// make form submittable
-		ThemesEditor.bindSaveIconEvent();
+		//ThemesEditor.bindSaveIconEvent();
```
Comment (MattiasNilsson) [inline Resources/Public/JavaScript/ThemesBackendModule.js]: Remove commented code

**Review by MattiasNilsson — COMMENTED** [review]
It looks good just some minor stuff.

File: Classes/Controller/EditorController.php [hunk 5]
```diff
@@ -54,6 +62,133 @@ class EditorController extends ActionController
...
+    {
+        /** @var BackendTemplateView $view */
+        parent::initializeView($view);
+
+        if ($this->view->getModuleTemplate() !== NULL) {
+            $pageRenderer = $this->view->getModuleTemplate()->getPageRenderer();
+            $pageRenderer->loadRequireJsModule('TYPO3/CMS/Themes/Colorpicker');
+            $pageRenderer->loadRequireJsModule('TYPO3/CMS/Themes/ThemesBackendModule');
+
+            $extRealPath = '../' . ExtensionManagementUtility::siteRelPath('themes');
```
Comment (MattiasNilsson) [inline Classes/Controller/EditorController.php]: $extRealPath = ExtensionManagementUtility::extPath('themes');

File: Classes/Controller/EditorController.php [hunk 6]
```diff
@@ -54,6 +62,133 @@ class EditorController extends ActionController
...
+    {
+        /** @var BackendTemplateView $view */
+        parent::initializeView($view);
+
+        if ($this->view->getModuleTemplate() !== NULL) {
+            $pageRenderer = $this->view->getModuleTemplate()->getPageRenderer();
+            $pageRenderer->loadRequireJsModule('TYPO3/CMS/Themes/Colorpicker');
+            $pageRenderer->loadRequireJsModule('TYPO3/CMS/Themes/ThemesBackendModule');
+
+            $extRealPath = ExtensionManagementUtility::siteRelPath('themes');
```
Comment (MattiasNilsson) [inline Classes/Controller/EditorController.php:98]: ExtensionManagementUtility::siteRelPath('themes');

needs to be changed to

ExtensionManagementUtility::extPath('themes');

Comment (anjeylink) [reply Classes/Controller/EditorController.php:98]: extPath return path /var/www/sitefolder/. i think this won't work for css files.

**Comment by MattiasNilsson** [comment]
Nice 👍

**Pull request merged**